oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -702,62 +756,66 @@ impl Runtime {
         with_writer_channel(channel_half, |channel| Ok(channel.label.clone()))
     }
 
-    /// Returns whether the given Node is allowed to read from the provided channel, according to
-    /// their respective [`Label`]s.
+    /// Returns whether the given Node is allowed to read from the provided channel read half,
+    /// according to their respective [`Label`]s.
     fn validate_can_read_from_channel(
         &self,
         node_id: NodeId,
         channel_half: &ChannelHalf,
     ) -> Result<(), OakStatus> {
-        trace!(
-            "{:?}: validating readability of {:?}",
-            node_id,
-            channel_half
-        );
-
-        let node_label = self.get_node_label(node_id);
         let channel_label = self.get_reader_channel_label(&channel_half)?;
+        self.validate_can_read_from_label(node_id, &channel_label)
+    }
+
+    /// Returns whether the given Node is allowed to read from an entity with the provided
+    /// [`Label`], taking into account any dowgrades it may be capable of.
+    fn validate_can_read_from_label(
+        &self,
+        node_id: NodeId,
+        label: &Label,
+    ) -> Result<(), OakStatus> {
+        let downgraded_node_label = self.get_node_downgraded_label(node_id);
         trace!(
-            "{:?}: node_label={:?}, channel_label={:?}",
+            "{:?}: can {:?} read from {:?}?",
             node_id,
-            node_label,
-            channel_label
+            downgraded_node_label,
+            label
```

Comment:
I had it there, but removed it because it increases the risk that in some future refactoring we would mix up the labels :) but I'll think about how to log more stuff in the future.